framework/src/auxkernels/FluxAux.C

```diff
@@ -0,0 +1,42 @@
+#include "FluxAux.h"
+
+template <>
+InputParameters
+validParams<FluxAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  // Declare the options for a MooseEnum.
+  // These options will be presented to the user in Peacock
+  // and if something other than these options is in the input file
+  // an error will be printed
+  MooseEnum component("x y z");
+
+  // Use the MooseEnum to add a parameter called "component"
+  params.addRequiredParam<MooseEnum>("component", component, "The desired component of flux.");
+
+  params.addRequiredCoupledVar("diffusion_variable",
+                               "The name of the variable");
+  params.addRequiredParam<MaterialPropertyName>(
+      "diffusivity",
+      "The name of the diffusivity material property that will be used in the flux computation.");
```

Comment:
Use `params.addClassDescription()` to add a short description of the class. 
You'll also need to add a Markdown file in moose/docs/content/documentation/systems/AuxKernels/framework to document this class. Use one of the existing ones in that directory as a template.